### PR TITLE
Fix featured title

### PR DESCRIPTION
**Pull request by mixcas**

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -293,6 +293,49 @@ Ajaxy = {
...
+  setCols: function() {
+    var _this = this;
+
+    if ( $('#featured-post-container').length ) {
+      
+      // Title
+      var $titleHolder = $('#featured-post-title-holder');
+      var titleCol = $titleHolder.attr('class').replace('col s','');
+      titleCol *= 1;
+      for (var i = titleCol; $titleHolder.width() < $('#featured-post-title').width() && i < 5; i++ ) {
```
Comment (varffs) [inline js/main.js]: does caching `$titleHolder` here stop the `.width()` from changing?

File: js/main.js [hunk 2]
```diff
@@ -293,6 +293,49 @@ Ajaxy = {
...
+        $titleHolder.removeClass('s' + i).addClass('s' + (i+1));
+        titleCol = i+1;
+      }
+
+      // Subtitle
+      var $subtitleHolder = $('#featured-post-subtitle-holder');
+      var subtitleCol = $subtitleHolder.attr('class').replace('col s','');
+      subtitleCol *= 1;
+      for (var i = subtitleCol; $subtitleHolder.width() < $('#featured-post-subtitle').width() && i < 4; i++ ) {
+        $subtitleHolder.removeClass('s' + i).addClass('s' + (i+1));
```
Comment (varffs) [inline js/main.js]: this maths with no spaces isn't linting well

**Comment by varffs** [comment]
why have u removed `.fix-widows`? it should be ok if set on that <a> or no?

File: front-page.php [hunk 3]
```diff
@@ -30,20 +30,21 @@
...
-          <h4 id="featured-post-subtitle" class="rotate-text font-condensed js-fix-widows">
+        <div id="featured-post-subtitle-holder" class="col s2">
+          <h4 id="featured-post-subtitle" class="rotate-text font-condensed">
             <a href="<?php the_permalink() ?>"><?php echo $subtitle; ?></a>
           </h4>
         </div>
     <?php } ?>
-        <div id="featured-post-image-holder">
+        <div class="col s1"></div>
+        <div id="featured-post-image-holder" class="col s19">
```
Comment (varffs) [inline front-page.php]: this should be `.s14` according to the design

File: front-page.php [hunk 4]
```diff
@@ -30,20 +30,21 @@
     $subtitle = get_post_meta( $post->ID, '_igv_post_subtitle', true );
   ?>
 
-      <article <?php post_class(); ?> id="featured-post-container">
-        <div id="featured-post-title-holder">
-          <h3 id="featured-post-title" class="rotate-text js-fix-widows">
+      <article <?php post_class('row'); ?> id="featured-post-container">
+        <div id="featured-post-title-holder" class="col s2">
+          <h3 id="featured-post-title" class="rotate-text">
```
Comment (varffs) [inline front-page.php]: the line height for both the `<h3>` and `<h4>` are wrong. The h3 needs less and the h4 more. Possibly the h3 is a little too big now. I can do this by eye once everything else is good

**Comment by varffs** [comment]
screen shot 2016-03-29 at 18 43 54

this happened. aside the widows thing looks like max height might need to be set via js to match the height of the image?

**Comment by mixcas** [comment]
@varffs yeah, i removed `.fix-widows` cuz it would still fucking things up. Like add a an empty line at the end of the title.

I think we could add it back and test it once eveything is fine.

**Comment by mixcas** [comment]
@varffs talking about setting the height with js.

I can't get to solve it in my mind. I was thinking this:
- get image height
- set title and subtitle height
- set columns on title, subtitle and image
  (at this point image height may have changed again)
  ??

I know it's not impossible ¯_(ツ)_/¯

Also, what size did that screenshot happen?

**Comment by varffs** [comment]
Na fix widows was fucking up cus it was set on the wrong element. the `<a>` was inside the `<h3>` and the heading had the fix widows which if you read the fix widows code makes some sense to not work. Or?

**Comment by varffs** [comment]
Why would the image height change? The image is going to be consistently the same size no? At least it should be as that is the one element we shouldn't need to touch. It is an image in a fixed ratio 14 columns wide.

File: js/main.js [hunk 5]
```diff
@@ -308,7 +308,15 @@ Featured = {
     var _this = this;
 
     if ( $('#featured-post-container').length ) {
+
+      // Get image height
+      $maxHeight = $('#featured-post-image-holder img').height();
```
Comment (varffs) [inline js/main.js]: this image has the id `featured-post-image`

File: js/main.js [hunk 6]
```diff
@@ -308,7 +308,15 @@ Featured = {
...
       
+      // Set max heights
+      if( $(window).width() > 720 ) {
```
Comment (varffs) [inline js/main.js:316]: why only above 720?

File: js/main.js [hunk 7]
```diff
@@ -308,7 +308,15 @@ Featured = {
...
+        $('#featured-post-container, #featured-post-title-holder, #featured-post-subtitle-holder, #featured-post-subtitle, #featured-post-image-holder').css('max-height', $maxHeight + 'px');
+      }
```
Comment (varffs) [inline js/main.js:320]: needs an else to explicitly set height to `auto` if the window is resized from e.g. 780 to 650

File: js/main.js (hunk 6 above)
Comment (mixcas) [inline js/main.js:316]: cuz thats the media-query size that changes the layout

**Pull request merged**